pkg/auth/providers/activedirectory/activedirectory_provider.go

```diff
@@ -125,14 +125,43 @@ func (p *adProvider) GetPrincipal(principalID string, token v3.Token) (v3.Princi
 		return v3.Principal{}, nil
 	}
 
+	lConn, err := p.ldapConnection(config, caPool)
+	if err != nil {
+		return v3.Principal{}, nil
+	}
+	defer lConn.Close()
+
 	parts := strings.SplitN(principalID, ":", 2)
 	if len(parts) != 2 {
 		return v3.Principal{}, errors.Errorf("invalid id %v", principalID)
 	}
+	var uniqueID string
 	scope := parts[0]
 	externalID := strings.TrimPrefix(parts[1], "//")
+	var distinguishedName string
+	if strings.EqualFold(UserScope, scope) {
+		if config.UserUniqueIDAttribute == "" {
```

Comment:
These new properties on the config: UserUniqueIDAttribute and GroupUniqueIDAttribute - are they editable? So can an admin go and change these values after AD auth is turned ON?
 If yes, what will happen when this property is not set initially - but set later on? The externalID for the users created earlier will be 'DN' and say then admin goes and sets these two properties. So any user accounts created after that will use the uniqueID as the externalID.

How will the code handle these two types of user accounts?